scripts/gen_docs.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -e
+
+# Author: Thomas Bechtold <tbechtold@suse.com>
+# License: Apache-2.0
+
+# List of project for running the doc generation
+: ${PROJECTS:="nova horizon cinder glance neutron manila aodh barbican ceilometer designate heat ironic magnum murano neutron-fwaas neutron-lbaas sahara"}
+# swift docs don't build successfully on stable/rocky
+# keystone docs dont build successfully with the script, requires python3, which should be whats being used
+# directory where the OpenStack git repositories are in
+: ${OPENSTACK_REPO_DIR:=${PWD}/devel/openstack}
+# branch to use when creating the doc
+: ${OPENSTACK_REPO_BRANCH:=stable/rocky}
+# extra tox options (eg. -r)
+: ${TOX_OPTS:=""}
+# the base output directory for the html files
+: ${OUTPUT_DIR:=${PWD}/openstack-docs-html}
+
+
+if [ ! -d "${OPENSTACK_REPO_DIR}" ]; then
+    mkdir -p ${OPENSTACK_REPO_DIR}
+fi
+
+if [ ! -d "${OUTPUT_DIR}" ]; then
+    mkdir -p ${OUTPUT_DIR}
+fi
+        
+for project in $PROJECTS; do
+    echo "##### ${project}"
+    (
+        if [ ! -d "${OPENSTACK_REPO_DIR}/${project}" ]; then
+            cd ${OPENSTACK_REPO_DIR}
+            echo "### Cloning git repo"
+            git clone https://git.openstack.org/openstack/${project}
+        fi
+        cd ${OPENSTACK_REPO_DIR}/${project}
+        git checkout ${OPENSTACK_REPO_BRANCH}
+        echo "### Clear out any changes that may have come from previous failed runs"
+        git checkout doc/source/conf.py
+        echo "### Updating conf.py"
+        sed -i "s/html_theme = 'openstackdocs'/html_theme = 'suse_sphinx_theme'/g" doc/source/conf.py
+        if [ -f doc/requirements.txt ]; then
+            echo "### Updating doc/requirements.txt"
+            git checkout doc/requirements.txt
+            grep -q -x -F 'suse_sphinx_theme' doc/requirements.txt || echo '-e git+https://github.com/SUSE/suse-sphinx-theme.git@master#egg=suse_sphinx_theme' >> doc/requirements.txt
+        else
+            if [ -f test-requirements.txt ]; then
+               echo "### Updating requirements.txt"
```

Comment:
Nit: this should say "Updating test-requirements.txt"

Reply:
fixed in latest version